app/src/main/java/de/westnordost/streetcomplete/quests/memorial_type/MemorialType.kt

```diff
@@ -0,0 +1,36 @@
+package de.westnordost.streetcomplete.quests.memorial_type
+
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+
+sealed interface MemorialTypeAnswer
+
+enum class MemorialType(val osmValue: String) : MemorialTypeAnswer {
+    STATUE("statue"),
+    BUST("bust"),
+    PLAQUE("plaque"),
+    WAR_MEMORIAL("war_memorial"),
+    STONE("stone"),
+    OBELISK("obelisk"),
+}
+
+enum class Stele(val osmMaterialValue: String) : MemorialTypeAnswer {
+    WOODEN_STELE("wood"),
+    STONE_STELE("stone"),
+    KHACHKAR_STELE("stone"),
+}
+
+fun MemorialTypeAnswer.applyTo(tags: Tags) {
+    when (this) {
+        is MemorialType -> {
+            tags["memorial"] = this.osmValue
+        }
+        is Stele -> {
+            tags["memorial"] = "stele"
+            tags["material"] = this.osmMaterialValue
+        }
+    }
+
+    if (this == Stele.KHACHKAR_STELE) {
+        tags["stele"] = "khachkar"
+    }
+}
```

Comment:
This would be better done like the stile quest (and others) which will make finishing/redoing my taginfo PR much easier:
https://github.com/streetcomplete/StreetComplete/blob/6aa3fb16b18f74955ce79a19b201d5bb60707616/app/src/main/java/de/westnordost/streetcomplete/quests/barrier_type/StileTypeAnswer.kt#L11-L16

```suggestion
enum class MemorialType(val osmValue: String, val osmMaterialValue: String? = null, val osmSteleValue: String? = null) : MemorialTypeAnswer {
    STATUE("statue"),
    BUST("bust"),
    PLAQUE("plaque"),
    WAR_MEMORIAL("war_memorial"),
    STONE("stone"),
    OBELISK("obelisk"),
    WOODEN_STELE("stele", "wood"),
    STONE_STELE("stele", "stone"),
    KHACHKAR_STELE("stele", "stone", "khachkar"),
}

fun MemorialTypeAnswer.applyTo(tags: Tags) {
    tags["memorial"] = this.osmValue
    if (this.osmMaterialValue != null) {
        tags["material"] = this.osmMaterialValue
    }
    if (this.osmSteleValue != null) {
        tags["stele"] = this.osmSteleValue
    }
}
```

Unless we can already (or maybe we should add) a way to add a tag if the value isn't null?

Reply:
I see this was resolved with no changes or comments, was there a reason behind that?

Even if the Khachkar Stele is going to be dropped, the principle can be easily adapted for the others.